api/types/events/resource_ids.go

```diff
@@ -32,3 +36,73 @@ func ResourceIDs(resourceIDs []types.ResourceID) []ResourceID {
 	}
 	return out
 }
+
+// ToEventResourceAccessIDs converts []types.ResourceAccessID to []ResourceAccessID for events.
+func ToEventResourceAccessIDs(in []types.ResourceAccessID) []ResourceAccessID {
+	out := make([]ResourceAccessID, 0, len(in))
+	for _, r := range in {
+		out = append(out, ToEventResourceAccessID(r))
+	}
+	return out
+}
+
+// ToEventResourceAccessID converts a types.ResourceAccessID to a ResourceAccessID for events.
+func ToEventResourceAccessID(in types.ResourceAccessID) ResourceAccessID {
+	out := ResourceAccessID{
+		Id: ResourceID{
+			ClusterName:     in.Id.ClusterName,
+			Kind:            in.Id.Kind,
+			Name:            in.Id.Name,
+			SubResourceName: in.Id.SubResourceName,
+		},
+	}
+
+	c := in.Constraints
+	if c == nil {
+		return out
+	}
+
+	details := c.GetDetails()
+	if details == nil {
+		return out
+	}
+
+	switch details.(type) {
+	// AWS Console constraints variant
+	case *types.ResourceConstraints_AwsConsole:
+		aws := c.GetAwsConsole()
+		if aws == nil {
+			// If payload is missing treat as unknown/unsupported
+			out.Constraints = &ResourceAccessID_UnknownConstraints{UnknownConstraints: &UnknownConstraints{}}
+			break
+		}
+
+		roleARNs := aws.RoleArns
+		count := len(roleARNs)
+		preview, truncated := previewStrings(roleARNs, MaxAuditRoleARNPreview)
+
+		out.Constraints = &ResourceAccessID_AwsConsole{
+			AwsConsole: &AWSConsoleConstraints{
+				RoleArnsCount:     uint32(count),
+				RoleArnsPreview:   preview,
+				RoleArnsTruncated: truncated,
+			},
+		}
+	// Unknown/unsupported constraint variant
+	default:
+		out.Constraints = &ResourceAccessID_UnknownConstraints{UnknownConstraints: &UnknownConstraints{}}
+	}
+
+	return out
+}
+
+// previewStrings returns up to limit elements from in, and a truncation flag
+func previewStrings(in []string, limit int) (preview []string, truncated bool) {
+	if limit <= 0 || len(in) == 0 {
+		return nil, false
+	}
+	if len(in) <= limit {
+		return append([]string(nil), in...), false
+	}
+	return append([]string(nil), in[:limit]...), false
```

Comment:
I guess you meant returning `true` here, right? And BTW, it would be great to see some tests to all of the newly defined functions from this file. (At least the public ones.)